src/caja-window.c

```diff
@@ -2170,7 +2185,18 @@ caja_window_class_init (CajaWindowClass *class)
     GtkBindingSet *binding_set;
 
     G_OBJECT_CLASS (class)->constructor = caja_window_constructor;
-    G_OBJECT_CLASS (class)->constructed = caja_window_constructed;
+#ifdef HAVE_WAYLAND
+    /*For some reason if we use the constructed() class method
+     *in wayland with the compositor set to or forcing CSD
+     *we lose the ability to drag the window by the titlebar
+     */
+    GdkDisplay *display = gdk_display_get_default();
+    if (GDK_IS_X11_DISPLAY (display))
```

Comment:
This should be the exact opposite test as in constructor(), so `! GDK_IS_WAYLAND_DISPLAY (display)`.  OK, it's likely not gonna matter anytime soon, but it's conceptually more correct, and could cause headaches if at some point there's more that just X11 vs. Wayland (does anybody remember the MIR backend? :smile: )

Reply:
This is no longer a factor as we now move this code to the constructor() function unconditionally